### ISSUE TITLE
SUMIFS ignoring blank cells in not-equal comparisons

### ISSUE THREAD

**Issue by AjayChauhan1982** [ClosedXML/ClosedXML#1567]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.95.3

**What is the current behavior?**

Blank cells are being ignored while evaluating formulas.
e.g: =SUMIFS(Sheet2!C3:C8, Sheet2!D3:D8,"<>B")

![image](https://user-images.githubusercontent.com/74641883/99542379-d876cf00-2966-11eb-9cba-175869339b5b.png)

in excel it is calculated as 17 but closedxml calculating it as 14 ignoring D5 which is blank. And when D8 is updated with value Alt-A, it throws error. 

**What is the expected behavior or new feature?**

Formula evaluation must include blank cells while comparing values.

**Is this a regression from the previous version?**

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    var filePath = @"";// attached file path (i.e FBT_BV.xlsx)
    var fileBytes = File.ReadAllBytes(filePath);

        using (MemoryStream ms = new MemoryStream(fileBytes))
        {
            var workbook = new XLWorkbook(ms);
            var sheet = workbook.Worksheet("Sheet1");
            var calculatedValue1 = sheet.Cell("B8").Value;//Expected Value 17
            var calculatedValue2 = sheet.Cell("C8").Value;//Expected Value 13
            var calculatedValue3 = sheet.Cell("D8").Value;//Expected Value 3
        }
}
```
[FBT_BV.xlsx](https://github.com/ClosedXML/ClosedXML/files/5560665/FBT_BV.xlsx)

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)

**Comment by AjayChauhan1982** [comment]
cc: @igitur

**Comment by vbjay** [comment]
RTFM
Reproducibility

*This is an important section. Read it carefully. Failure to do so will
cause a 'RTFM' comment.*

Without a code sample, it is unlikely that your issue will get attention.
Don't be lazy. Do the effort and assist the developers to reproduce your
problem. Code samples should be minimal complete and verifiable
<https://stackoverflow.com/help/mcve>. Sample spreadsheets should be
attached whenever applicable. Remove sensitive information.

*Code to reproduce problem:*

public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code
    //   samples on the wiki are good examples of the terseness that I
want. Don't
    //   post your full application.
}


   -  I attached a sample spreadsheet. (You can drag files on to this issue)

**Comment by AjayChauhan1982** [comment]
static void Main(string[] args)
        {
            var filePath = @"";// attached file path (i.e FBT_BV.xlsx)
            var fileBytes = File.ReadAllBytes(filePath);
            
            using (MemoryStream ms = new MemoryStream(fileBytes))
            {
                var workbook = new XLWorkbook(ms);
                var sheet = workbook.Worksheet("Sheet1");
                var calculatedValue1 = sheet.Cell("B8").Value;//Expected Value 17
                var calculatedValue2 = sheet.Cell("C8").Value;//Expected Value 7
                var calculatedValue3 = sheet.Cell("D8").Value;//Expected Value 3
            }
               
        }

[FBT_BV.xlsx](https://github.com/ClosedXML/ClosedXML/files/5560555/FBT_BV.xlsx)

**Comment by AjayChauhan1982** [comment]
Hi, 

In closedXML followin code breaks the expression into wrong expression if value contains any special character.
e.g: if value is "Alt-A" and criteria is "<>Test" , so correct expression should be "Alt-A"<>"Test" but when following code executed it truncate the expression into "Alt"-"A". Please do the needful.

//copied from ValueSatisfiesCriteria() method inside CalcEngineHelpers.cs class
 
                    // add quotes if necessary
                    var pattern = @"([\w\s]+)(\W+)(\w+)";
                    var m = Regex.Match(expression, pattern);
                    if (m.Groups.Count == 4
                        && (!double.TryParse(m.Groups[1].Value, out double d) ||
                            !double.TryParse(m.Groups[3].Value, out d)))
                    {
                        expression = string.Format("\"{0}\"{1}\"{2}\"",
                                                   m.Groups[1].Value,
                                                   m.Groups[2].Value,
                                                   m.Groups[3].Value);
                    }


cc @igitur , @vbjay , @ChiragRupani

**Comment by jahav** [comment]
SUMIFS has been reimplemented in #2499, but it still had a the issue described here. That was fixed in #2510.

I have tried the provided reproducible example and it calculates correct values (I have force reevaluation, because workbook contained precalculated values).
